### ISSUE TITLE
Failed to close monitoring services due to NPE

### ISSUE THREAD

**Issue by Zelldon** [camunda-cloud/zeebe#7854]
**Describe the bug**

The nighly testbench run found in our snapshot version an problem in closing the broker. The error occurred 15 times in the nightly run. It seems that a NPE causes to `Shutdown failed in the following steps: [monitoring services]`

Error group: https://console.cloud.google.com/errors/CPeYl8DgzYEO?service=zeebe&time=P7D&refresh=off&project=camunda-cloud-240911

**To Reproduce**

Run the snapshot I assume. One error I saw occurred when a node became leader 1 minute before and got closed afterwards.

**Expected behavior**

No errors on closing.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.util.concurrent.ExecutionException: Shutdown failed in the following steps: [monitoring services]. See suppressed exceptions for details.
	at io.camunda.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:141) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:108) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.FutureUtil.join(FutureUtil.java:21) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.future.CompletableActorFuture.join(CompletableActorFuture.java:196) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.Broker.lambda$migratedStartupSteps$7(Broker.java:202) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.bootstrap.CloseProcess.lambda$closingStepByStep$0(CloseProcess.java:59) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:93) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.bootstrap.CloseProcess.closingStepByStep(CloseProcess.java:59) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:93) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.bootstrap.CloseProcess.closeReverse(CloseProcess.java:40) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.Broker.lambda$close$14(Broker.java:395) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at java.util.concurrent.CompletableFuture.uniAcceptNow(Unknown Source) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniAcceptStage(Unknown Source) ~[?:?]
	at java.util.concurrent.CompletableFuture.thenAccept(Unknown Source) ~[?:?]
	at io.camunda.zeebe.broker.Broker.lambda$close$15(Broker.java:393) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.LogUtil.doWithMDC(LogUtil.java:23) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.Broker.close(Broker.java:388) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.StandaloneBroker.onApplicationEvent(StandaloneBroker.java:80) ~[camunda-cloud-zeebe-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.StandaloneBroker.onApplicationEvent(StandaloneBroker.java:31) ~[camunda-cloud-zeebe-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.broker.StandaloneBroker$$EnhancerBySpringCGLIB$$1e18863e.onApplicationEvent(<generated>) ~[camunda-cloud-zeebe-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176) ~[spring-context-5.3.10.jar:5.3.10]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:169) ~[spring-context-5.3.10.jar:5.3.10]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143) ~[spring-context-5.3.10.jar:5.3.10]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:421) ~[spring-context-5.3.10.jar:5.3.10]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:378) ~[spring-context-5.3.10.jar:5.3.10]
	at org.springframework.context.support.AbstractApplicationContext.doClose(AbstractApplicationContext.java:1058) ~[spring-context-5.3.10.jar:5.3.10]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.doClose(ServletWebServerApplicationContext.java:172) ~[spring-boot-2.5.4.jar:2.5.4]
	at org.springframework.context.support.AbstractApplicationContext.close(AbstractApplicationContext.java:1021) ~[spring-context-5.3.10.jar:5.3.10]
	at org.springframework.boot.SpringApplicationShutdownHook.closeAndWait(SpringApplicationShutdownHook.java:131) ~[spring-boot-2.5.4.jar:2.5.4]
	at java.lang.Iterable.forEach(Unknown Source) [?:?]
	at org.springframework.boot.SpringApplicationShutdownHook.run(SpringApplicationShutdownHook.java:100) [spring-boot-2.5.4.jar:2.5.4]
	at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: io.camunda.zeebe.util.startup.StartupProcessException: Shutdown failed in the following steps: [monitoring services]. See suppressed exceptions for details.
	at io.camunda.zeebe.util.startup.StartupProcess.aggregateExceptionsSynchronized(StartupProcess.java:282) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.startup.StartupProcess.completeShutdownFutureSynchronized(StartupProcess.java:265) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.startup.StartupProcess.proceedWithShutdownSynchronized(StartupProcess.java:230) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.startup.StartupProcess.lambda$proceedWithShutdownSynchronized$6(StartupProcess.java:250) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:33) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:76) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:94) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:78) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:191) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	Suppressed: io.camunda.zeebe.util.startup.StartupProcessStepException: Bootstrap step monitoring services failed
		at io.camunda.zeebe.util.startup.StartupProcess.lambda$proceedWithShutdownSynchronized$6(StartupProcess.java:244) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.sched.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:33) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:76) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:94) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:78) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:191) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	Caused by: java.lang.NullPointerException
		at io.camunda.zeebe.broker.bootstrap.BrokerStepMetricDecorator.callDelegateAndUpdateTimer(BrokerStepMetricDecorator.java:61) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.broker.bootstrap.BrokerStepMetricDecorator.shutdown(BrokerStepMetricDecorator.java:51) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.broker.bootstrap.BrokerStepMetricDecorator.shutdown(BrokerStepMetricDecorator.java:25) ~[zeebe-broker-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.startup.StartupProcess.proceedWithShutdownSynchronized(StartupProcess.java:236) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.startup.StartupProcess.lambda$proceedWithShutdownSynchronized$6(StartupProcess.java:250) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		at io.camunda.zeebe.util.sched.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:28) ~[zeebe-util-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
		... 6 more
```

</p>
</details>

**Environment:**
 - OS: 
 - Zeebe Version: SNAPSHOT
 - Configuration:

**Comment by Zelldon** [comment]
Occurred as well in our medic benchmark 

https://console.cloud.google.com/errors/CPeYl8DgzYEO?service=zeebe-broker&version=medic-cw-38-bd7cc81288-benchmark&time=P7D&project=zeebe-io

**Comment by pihme** [comment]
Fixed with #7888 on develop and #7890 in release branch
